AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/logger/appender/MessagerAppender.java

```diff
@@ -66,4 +85,110 @@ private Kind resolveKind(Level level) {
 		return Kind.OTHER;
 	}
 
+	private class Message {
+		private final Kind kind;
+		private final String message;
+		private final String annotationMirrorString;
+		private final List<String> elements = new LinkedList<>();
+
+		Message(Kind kind, String message, Element element, AnnotationMirror annotationMirror) {
+			this.kind = kind;
+			this.message = message;
+			this.annotationMirrorString = annotationMirror == null ? null : annotationMirror.toString();
+
+			if (element != null) {
+				Element enclosingElement = element;
+				do {
+					elements.add(0, enclosingElement.toString());
+					enclosingElement = enclosingElement.getEnclosingElement();
+				} while (!enclosingElement.getKind().equals(ElementKind.PACKAGE));
+			}
+		}
+
+		private AnnotationMirror getAnnotationMirror(Element element) {
+			if (element == null) {
+				return null;
+			}
+
+			for (AnnotationMirror mirror : element.getAnnotationMirrors()) {
+				if (mirror.toString().equals(annotationMirrorString)) {
+					return mirror;
+				}
+			}
+			return null;
+		}
+
+		private Element getElement() {
```

Comment:
This is a rather complicated method, should not we cache the result? Or this is called only once?

Reply:
I think we should move this to an utility method, this logic is too much for a `getter`.

Reply:
As far as i can see it is only called once when the appender is closed in the last round.

Reply:
what do you mean by `move this to an utility method`? If it is to much logic for a "getter" i'd just rename the method to something like `findRelatedElement` or so. :D

Reply:
Okay, we can leave this here.